include/xtensor/views/xslice.hpp

```diff
@@ -1380,8 +1331,9 @@ namespace xt
     }
 
     template <class T>
-    template <class S, typename>
+    template <class S>
     inline xkeep_slice<T>::operator xkeep_slice<S>() const noexcept
+        requires std::convertible_to<S, T>
```

Comment:
Same here (sorry for the exhaustive list, but I find it easier to fix them all when going through the review).